### PR TITLE
Restructure docs files for new contributing guideline

### PR DESCRIPTION
**Pull request by samredai**
(This PR should be reviewed as a pair with [apache/iceberg-docs/pull/59](https://github.com/apache/iceberg-docs/pull/59))

TLDR;
- `./docs/common/format`: moved to `./format`
- `./docs/common`: removed (will only be maintained in apache/iceberg-docs going forward)
- `./docs/versioned`: contents moved up one level to simple `./docs`
- `./docs/rest`: moved to `./open-api`

### Additional context
This restructures the docs files for the new split contribution model. The updated instructions can be found in the `apache/iceberg-docs` repo, but to summarize: PR's for common pages will go against `apache/iceberg-docs`, and PR's for versioned pages will go against `apache/iceberg`.

Therefore this PR removes the common pages (except for those under `format` such as `spec.md` which will be maintained here). This also flattens the `docs` directory so that it matches what the release manager will need to copy over (just copy the `docs` folder here to `docs` folder in the apache/iceberg-docs repo).

The rest spec has been moved to a root level `open-api` directory.

### CODE REVIEW

**Comment by rdblue** [comment]
@jackye1995 FYI. This restructures docs so that only versioned docs and spec docs are maintained here. The other docs are in the iceberg-docs repo. We think this is easier because we found that a lot of PRs in this repo had to be manually copied over and contributors found it confusing that the source for the common docs were kept in two places.

**Comment by rdblue** [comment]
Looks good to me, but it would be good to hear from @jackye1995 before merging.

**Review by kbendick — APPROVED** [review]
LGTM.

**Comment by rdblue** [comment]
@samredai, can you rebase? Looks like there are conflicts now.

**Comment by samredai** [comment]
Thanks for the ping @rdblue. Rebased. 👍

**Comment by rdblue** [comment]
It's out of date again...

**Comment by samredai** [comment]
@rdblue thanks again, rebased :)

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Looks like we need to turn off CI for `docs/` paths. @kbendick, can you open a PR?

**Comment by kbendick** [comment]
> Looks like we need to turn off CI for `docs/` paths. @kbendick, can you open a PR?

Yeah will do so now. There are a few other CI related  PRs in the pipeline I can help rebase etc if need be.

**Comment by kbendick** [comment]
This should cover it. The `docs` folder was already present, but `open-api` and `format` were not: https://github.com/apache/iceberg/pull/4286
